### PR TITLE
alerting: notes on label matchers

### PR DESCRIPTION
**Pull request by tonypowa**
**What is this feature?**

- warns the user that extra space in templated labels can break the match with the policy
- highlights that all alert labels must match the policy labels
- clarifies how matching works when multiple labels are present 

**Reference**:

https://github.com/grafana/grafana/issues/89203

https://community.grafana.com/t/grafana-alert-does-not-route-to-1-notification-policy/124034/2

### CODE REVIEW

**Comment by tonypowa** [comment]
testing locally

**Comment by tonypowa** [comment]
/templates/

image


/notification-policies/

image

image
all good
let me know if it looks good @brendamuir  😃 

/templating-labels-annotations/

image

File: docs/sources/alerting/alerting-rules/templating-labels-annotations.md [hunk 1]
```diff
@@ -32,6 +32,10 @@ All templates should be written in [text/template](https://pkg.go.dev/text/templ
 
 Each template is evaluated whenever the alert rule is evaluated, and is evaluated for every alert separately. For example, if your alert rule has a templated summary annotation, and the alert rule has 10 firing alerts, then the template will be executed 10 times, once for each alert. You should try to avoid doing expensive computations in your templates as much as possible.
 
+{{% admonition type="caution" %}}
+Extra whitespace in label templates can break the match with notification policies.
```
Comment (brendamuir) [inline docs/sources/alerting/alerting-rules/templating-labels-annotations.md:36]:
```suggestion
Extra whitespace in label templates can break matches with notification policies.
```

File: docs/sources/alerting/fundamentals/notifications/templates.md [hunk 2]
```diff
@@ -106,6 +106,10 @@ Here are some commonly used built-in [variables](ref:variables-label-annotation)
 
         CPU usage for instance1 has exceeded 80% for the last 5 minutes: 81.2345
 
+{{% admonition type="caution" %}}
+Extra whitespace in label templates can break the match with notification policies.
```
Comment (brendamuir) [inline docs/sources/alerting/fundamentals/notifications/templates.md:110]:
```suggestion
Extra whitespace in label templates can break matches with notification policies.
```

**Review by brendamuir — APPROVED** [review]

**Pull request merged**